src/components/Menu-new/Menu.tsx

```diff
@@ -0,0 +1,112 @@
+import { ReactElement, useCallback, useState } from 'react';
+import { Icon, View, Text } from '@aws-amplify/ui-react';
+import Link from 'next/link';
+import { directory } from './buildDirectory.mjs';
+import styles from './Menu.module.scss';
+import { useRouter } from 'next/router';
+
+export function Menu(): ReactElement {
+  const rootPage = directory;
+  const platformOverviewPage = rootPage.children[0];
+
+  return (
+    <nav className={styles['menu']}>
+      <View marginBottom="16px">
+        <Text fontSize="16px" fontWeight="700">
+          How Amplify works
+        </Text>
+      </View>
+      <ul>
+        {platformOverviewPage.children &&
+          platformOverviewPage.children.map((child) => {
+            return generateListItem(child, 1);
+          })}
+      </ul>
+    </nav>
+  );
+}
+
+const CATEGORY_LEVEL = 1;
+
+function generateListItem(pageNode, level): ReactElement {
+  const router = useRouter();
+  const [open, setOpen] = useState(false);
+  const toggleDrawer = useCallback(() => {
+    setOpen((prevOpen) => !prevOpen);
+  }, []);
+
+  if (!pageNode) {
+    return <></>;
+  }
+
+  const pathname = `/${router.query.platform}/${pathHelper(pageNode.route)}/`;
+  console.log(pathname);
+  console.log('router', router);
+  const current = router.asPath === pathname;
+  console.log(current);
+
+  return (
```

Comment:
Thanks for calling this out! I'm looking into this at the moment - just to think out loud, wrapping this section in that check will make it so that the menu doesn't show the page when it's not supported.

I'm doing something like this:
```tsx
if (pageNode.platforms.includes(router.query.platform)) {
  // return the menu item element here
} else {
  return <></>;
}
```

When we navigate to a 404 though, this breaks the menu because the `router.query.platform` is undefined for the 404 page and so this `if` check just returns `<></>` and the menu doesn't get to recursively build itself. I need to somehow get it build the entire menu on the 404 page when there is no `platform`. Still thinking about it 🤔